src/modules/ens/ens.service.ts

```diff
@@ -294,4 +299,11 @@ export class EnsService {
     }
     this.logger.info('### Finished ENS Sync ###');
   }
+
+ onModuleDestroy() {
+    this.ensRegistry.removeAllListeners('Transfer');
+    this.ensRegistry.removeAllListeners('NewOwner');
+    this.domainNotifer.removeAllListeners('DomainUpdated');
+    this.publicResolver.removeAllListeners('TextChanged');
+  }
```

Comment:
This seems like a good idea, but just curious as to what triggered you to add it?
Also, not sure if this is possible, but it could be nice to add the listeners to a list when adding them and then this method could just loop through that list. In this way, if a new listener is added, then it will be automatically removed on destroy instead of needing to remember to add it. Not a necessary change though, just an idea 💡

Reply:
Sometimes ago when working in the staking tests, I discovered that test were timing out due to the service still listening for events even after the tests ended. This was a fix for that issue, thought it would be a good idea to replicate here too.